sources/packages/api/src/esdc-integration/e-cert-full-time-integration/e-cert-full-time-integration.service.ts

```diff
@@ -211,12 +215,92 @@ export class ECertFullTimeIntegrationService extends SFTPIntegrationBase<void> {
     } as CreateRequestFileNameResult;
   }
 
+  /**
+   * Get the list of all E-Cert response files from the
+   * MSFT-Response folder
+   * Filename pattern is `PEDU.PBC.CERTSFB.yyyymmdd.001`
+   * @returns full file paths for all response files from the
+   * E-Cert response folder.
+   */
+  async getResponseFilesFullPath(): Promise<string[]> {
+    let filesToProcess: Client.FileInfo[];
+    const client = await this.getClient();
+    try {
+      filesToProcess = await client.list(
+        `${this.esdcConfig.ftpResponseFolder}`,
+        new RegExp(
+          `^${this.esdcConfig.environmentCode}EDU.PBC.CERTSFB.*\\.DAT`,
+          "i",
+        ),
+      );
+    } finally {
+      await SshService.closeQuietly(client);
+    }
+    return filesToProcess.map(
+      (file) => `${this.esdcConfig.ftpResponseFolder}/${file.name}`,
+    );
+  }
+
+  // Generate Record
+
   /**
    * Transform the text lines in parsed objects specific to the integration process.
-   * TODO: read the feedback file.
    * @param remoteFilePath full remote file path with file name.
+   * @returns Parsed records from the file.
    */
-  downloadResponseFile(remoteFilePath: string): Promise<void> {
-    throw new Error("To be implemented.");
+  async downloadResponseFile(
+    remoteFilePath: string,
+  ): Promise<ECertResponseRecord[]> {
+    const fileLines = await this.downloadResponseFileLines(remoteFilePath);
+    /**
+     * Read the first line to check if the header code is the expected one.
+     * and remove header.
+     */
+    const header = ECertFileHeader.createFromLine(fileLines.shift());
+    if (header.recordTypeCode !== RecordTypeCodes.ECertHeader) {
+      this.logger.error(
+        `The E-Cert file ${remoteFilePath} has an invalid transaction code on header: ${header.recordTypeCode}`,
```

Comment:
I would recommend changing "transaction code" to "record type code" to be more aligned with the documentation.

Reply:
It seems that the file still has a couple of "transaction code". Can you check it, please?

Reply:
Thanks for updating it.